dot/parachain/prospective-parachains/prospective-parachains_test.go

```diff
@@ -0,0 +1,276 @@
+// correctly handles the case where there are no active leaves.
+func TestGetMinimumRelayParents_NoActiveLeaves(t *testing.T) {
+	mockView := &View{
+		activeLeaves:   map[common.Hash]bool{},
+		PerRelayParent: map[common.Hash]*RelayParentData{},
+	}
+
+	// Initialize ProspectiveParachains with the mock view
+	pp := &ProspectiveParachains{
+		View: mockView,
+	}
+
+	// Create a channel to capture the output
+	sender := make(chan []ParaIDBlockNumber, 1)
+
+	// Execute the method under test
+	pp.getMinimumRelayParents(common.BytesToHash([]byte("active_hash")), sender)
+	// Validate the results
+	result := <-sender
+	assert.Empty(t, result, "Expected result to be empty when no active leaves are present")
+}
+
+func TestProspectiveParachains_HandleMinimumRelayParents(t *testing.T) {
+	candidateRelayParent := common.Hash{0x01}
+	paraId := parachaintypes.ParaID(1)
+	parentHead := parachaintypes.HeadData{
+		Data: bytes.Repeat([]byte{0x01}, 32),
+	}
+	headData := parachaintypes.HeadData{
+		Data: bytes.Repeat([]byte{0x02}, 32),
+	}
+	validationCodeHash := parachaintypes.ValidationCodeHash{0x01}
+	candidateRelayParentNumber := uint32(0)
+
+	candidate := makeCandidate(
+		candidateRelayParent,
+		candidateRelayParentNumber,
+		paraId,
+		parentHead,
+		headData,
+		validationCodeHash,
+	)
+
+	subsystemToOverseer := make(chan any)
+	overseerToSubsystem := make(chan any)
+
+	prospectiveParachains := NewProspectiveParachains(subsystemToOverseer)
+
+	relayParent := inclusionemulator.RelayChainBlockInfo{
+		Hash:        candidateRelayParent,
+		Number:      0,
+		StorageRoot: common.Hash{0x00},
+	}
+
+	baseConstraints := &inclusionemulator.Constraints{
+		RequiredParent:       parachaintypes.HeadData{Data: bytes.Repeat([]byte{0x01}, 32)},
+		MinRelayParentNumber: 0,
+		ValidationCodeHash:   validationCodeHash,
+		MaxPoVSize:           1000000,
+	}
+
+	scope, err := fragmentchain.NewScopeWithAncestors(relayParent, baseConstraints, nil, 10, nil)
+	assert.NoError(t, err)
+
+	prospectiveParachains.View.PerRelayParent[candidateRelayParent] = &RelayParentData{
+		fragmentChains: map[parachaintypes.ParaID]*fragmentchain.FragmentChain{
+			paraId: fragmentchain.NewFragmentChain(scope, fragmentchain.NewCandidateStorage()),
+		},
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	var wg sync.WaitGroup
+
+	// Run the subsystem
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		prospectiveParachains.Run(ctx, overseerToSubsystem)
+	}()
+
+	sender := make(chan []ParaIDBlockNumber, 1)
+	go func() {
+		overseerToSubsystem <- GetMinimumRelayParents{
+			RelayChainBlockHash: candidateRelayParent,
+			Sender:              sender,
+		}
+	}()
+
+	result := <-sender
+	assert.Len(t, result, 1, "Expected one ParaIDBlockNumber in the result")
+	assert.Equal(t, paraId, result[0].ParaId, "ParaId mismatch in the result")
+	assert.Equal(t, uint32(7), result[0].BlockNumber, "BlockNumber mismatch in the result")
```

Comment:
why the resultant block number is expected to be equal 7 if the relay parent in the scope has number 0?

Reply:
@EclesioMeloJunior fixed